app/api/predictions/controller.py

```diff
@@ -14,202 +15,224 @@
 class PredictionRequest(BaseModel):
     prediction: str
     address: str
-    team:str
-
-
-async def get_api_key(request: Request) -> str:
-    """
-    Extracts the API key from the request headers and checks its validity.
-
-    Parameters:
-    request (Request): The incoming request object.
-
-    Returns:
-    str: The API key extracted from the headers.
-
-    Raises:
-    HTTPException: If the API key is invalid or missing.
-    """
-    api_key = request.headers.get("api_key_auth")
-    check_api_key(api_key)
-    return api_key
-
-
-@router.post("/", response_model=Union[Dict[str, Dict[str, str]], Dict[str, str]])
+    team: str
+
+
+class PredictionController(BaseController):
+    prediction_service = PredictionService()
+
+    @staticmethod
+    async def create_prediction(
+        request: PredictionRequest,
+    ) -> dict[str, dict] | dict[str, str]:
```

Comment:
Duplicated return type? `dict[str, dict] | dict[str, str]`

Reply:
yes typo, it's fixed